### DISCUSSION TITLE
Help with VUE Custom Components

### DISCUSSION THREAD

**Discussion by ag84ark** [area17/twill#877] [General]
I am trying to create a Vue Custom Component but I can't seem to get it work, looked all over and no idea how to do it.
I have created a component (.vue file) and added it in 'assets/js/components'.
Then I have run the 

> php artisan twill:build

 command but still not working.
Then I have runner `php artisan twill:dev` and in chrome devTools i get the error: 

> Unknown custom element : `<custom-event-attendees-table>` - did you register the component correctly? For recursive components, make sure to provide the "name" option.

The component has the "name" property, and it's set to: "CustomEventAttendeesTable", so I don't think the problem is from here.

I have looked into main-form.js and I see that the component is included in the file but just not working.
Do I have to declare the component somewhere else?

Another big problem that I have at the moment is that the custom components only are included in the main-form.js build but not into the main-free.js build which forces you to use the form view instead of the free one.

Thanks for helping!

**Comment by ag84ark** [comment]
I think I have found part of the problem, I need to put a17- before the component name.
Is there a possibility that in the main-free.js in the twill source to add this

```
// Custom form components
const importedComponents = require.context('@/components/customs/', true, /\.(js|vue)$/i)
importedComponents.keys().map(block => {
  const blockName = block.match(/\w+/)[0].replace(/([a-z])([A-Z])/g, '$1-$2').replace(/\s+/g, '-').toLowerCase()
  return Vue.component('a17-' + blockName, importedComponents(block).default)
})
```

so that custom components can be available in FREE layout?
Thanks.

**Comment by Insperio** [comment]
Yes I agree with this. 
I just re-worked most of the hacky attempts I used in older versions and this is only bit that still forced me to mess with twill vendor folder. 
It would be great to add it. Unless there is a different way to load custom components into those pages ?
